cmd/src/gateway_benchmark.go

```diff
@@ -152,7 +178,30 @@ func benchmarkEndpoint(client *http.Client, url string) time.Duration {
 		fmt.Printf("Error reading response body: %v\n", err)
 		return 0
 	}
+	if resp.StatusCode != http.StatusOK {
+		fmt.Printf("non-200 response: %v\n", resp.Status)
+		return 0
+	}
+
+	return time.Since(start)
+}
 
+func benchmarkEndpointWebSocket(conn *websocket.Conn) time.Duration {
```

Comment:
Naming: I'd have called this `benchmarkEndpointWebsocket` with a non-capital "s". I looked up the Go docs and your version seems more idiomatic with Go.
If we agree that "WebSocket" is the correct capitalization for entity names in Go then please also rename `gatewayWebsocket` and `sourcegraphWebsocket` accordingly. (I'd do it to fix my mistake but I don't want to mess or conflict with your PR.)

Reply:
I've changed it to "WebSocket" everywhere in https://github.com/sourcegraph/sourcegraph/pull/1526

Reply:
wikipedia says WebSocket so yea let's go with that capitalization

Reply:
I was aware of that, I just wasn't aware how to translate MixedCase entity names into Go entity names. But yeah, all clear, WebSocket it is!